text/0113-keyspace-level-gc.md

```diff
@@ -0,0 +1,83 @@
+# Keyspace level GC
+
+## Summary
+
+- RFC PR: https://github.com/tikv/rfcs/pull/113
+- Tracking Issue: https://github.com/tikv/tikv/issues/16896
+
+TiKV support keyspace level GC.
+
+## Motivation
+
+Previously, TiDB has supported the deployment of multiple TiDB clusters with different keyspaces 
+on a single PD TiKV cluster.
+
+We've implemented multiple TiDB clusters, with one global TiDB GC worker (A TiDB server without Keyspace configuration) 
+to calculate the global GC safe point and resolve locks, While each keyspace's TiDB has their own GC Worker, 
+keyspace GC Worker use the global GC safe point to do deleteRange in the sepical keyspace ranges.
+
+But old implementation causes the calculation of the global GC to depend on the oldest safe point and min start ts 
+of all keyspaces and TiDB cluster without keyspace configured.
+
+So we propose the implementation of keyspace level GC:
+
+TiDB PR https://github.com/pingcap/tidb/pull/51300 implements: 
+Isolation of GC safe point calculations between keyspaces (the concept is 'keyspace level GC'). 
+When the Keyspace GC safe point calculation of Keyspace with keyspace level GC is slow, 
+it will not affect other keyspaces GC safe point calculation.
+Keyspaces can be created by setting gc_management_type = keyspace_level_gc to enable keyspace level GC,
+then this keyspace can calculate GC safe point by itself.
+
+TiKV PR https://github.com/tikv/tikv/pull/16808 implemented on TiKV side: 
+In GC process, it parses the keyspace id from the data key, 
+combines the keyspace meta config and the keyspace level GC safe point corresponding to the keyspace id to determine
+the GC safe point value of the data key and execute the GC logic.
+
+
+## Concepts of GC management type:
+
+1. Global GC:
+    - Represents the previous default GC logic; there is a TiDB calculate the global GC safe point for the whole cluster.
+    - The default GC management type for keyspace is Global GC,
+2. Keyspace level GC:
+    - Indicates that the keyspace will advance its own GC safe point.
+    - It is possible and only possible to set gc_management_type = keyspace_level_gc when PD creates keyspaces.
+    - The keyspace which already set gc_management_type = keyspace_level_gc, 'gc_management_type' it can not be updated 
```

Comment:
Because currently gc_management_type changes are not supported after keyspaces are created, changing gc_management_type requires a more complex process.